compiler/graindoc/jsoo_hacks.js

```diff
@@ -1,28 +1,11 @@
-//Provides: caml_make_path
-//Requires: caml_current_dir
-//Requires: caml_jsstring_of_string
-//Requires: caml_root, jsoo_mount_point, MlNodeDevice
-if (caml_root.charCodeAt(1) == 58 && caml_root[0] != 'C') {
-  // This ensure we can find our C:/snapshot/ stuff even when running in a diff drive
-  // Not the ideal place to do this but it is the easiest to avoid free variables
-  jsoo_mount_point.push({ path: 'C:/', device: new MlNodeDevice('C:/') });
-}
```

Comment:
Why not keep the 5 lines above, they seem much simpler than the new logic split between js and ocaml.

Reply:
@hhugo they get removed from the final JSOO code. Am I doing something wrong? It'd be much nicer to keep these if they weren't removed.

Reply:
Code without a `//Provides` should no be dropped.

Reply:
🤔 In my testing, it seems like any `//Provides` that is not used as an `external` in the OCaml code gets dropped.

Reply:
Don't include the provide !
just put
```
//Requires: caml_root, jsoo_mount_point, MlNodeDevice
if (caml_root.charCodeAt(1) == 58 && caml_root[0] != 'C') {
  // This ensure we can find our C:/snapshot/ stuff even when running in a diff drive
  // Not the ideal place to do this but it is the easiest to avoid free variables
  jsoo_mount_point.push({ path: 'C:/', device: new MlNodeDevice('C:/') });
}
```
nothing else

Reply:
I'll give that a try!

Reply:
@hhugo Trying your example, the JS crashes with `caml_root is not defined`

Reply:
We should fix this. In the meantime, you could just keep theses lines under a `init_jsoo_for_grain` function (with a //Provides) and declare and call the external on the ml side.

Reply:
I was hoping I could do the `external` trick, but it seems that unless I provide a C-stub that it won't actually be detected (as per https://github.com/ocsigen/js_of_ocaml/issues/804#issuecomment-495826520). I tried providing the C-stub but it doesn't work because we have separate compilation targets for the JS vs native builds.

Reply:
Would you be open to adding an optional parameter to `Js_of_ocaml.Sys_js.mount` that specifies which filesystem type to mount? We would no longer need the separate JS file if that could be specified in OCaml.

Reply:
> I was hoping I could do the `external` trick, but it seems that unless I provide a C-stub that it won't actually be detected (as per [ocsigen/js_of_ocaml#804 (comment)](https://github.com/ocsigen/js_of_ocaml/issues/804#issuecomment-495826520)). I tried providing the C-stub but it doesn't work because we have separate compilation targets for the JS vs native builds.

I don't understand why this is an issue. Can't you have a c stub that does nothing ?

Reply:
This is the dune error that I got when trying to add the empty C-stub:
```
Error: Pure bytecode executables cannot contain foreign stubs.
Hint: If you only need to build a native executable use "(modes exe)".
```

Reply:
I've updated this PR to show the usage if we add the optional `fs_type` argument to `Sys_fs.mount`

Reply:
> We should fix this. In the meantime, you could just keep theses lines under a `init_jsoo_for_grain` function (with a //Provides) and declare and call the external on the ml side.

https://github.com/ocsigen/js_of_ocaml/issues/1141